File: scratch/read_docx.js
const fs = require('fs');
const unzipper = require('unzipper');

async function extractText(docxPath) {
    try {
        const directory = await unzipper.Open.file(docxPath);
        const file = directory.files.find(d => d.path === 'word/document.xml');
        if (!file) {
            console.log('document.xml not found');
            return;
        }
        const content = await file.buffer();
        const text = content.toString('utf-8');
        // Simple regex to extract text between <w:t> tags
        const regex = /<w:t[^>]*>([^<]+)<\/w:t>/g;
        let match;
        const result = [];
        while ((match = regex.exec(text)) !== null) {
            result.push(match[1]);
        }
        fs.writeFileSync('./scratch/docx_text.txt', result.join('\n'));
        console.log('Extraction complete');
    } catch (e) {
        console.error(e);
    }
}

extractText(process.argv[2]);

File: scratch/read_docx.py
import zipfile
import xml.etree.ElementTree as ET
import sys

def extract_text_from_docx(docx_path):
    try:
        with zipfile.ZipFile(docx_path) as docx:
            xml_content = docx.read('word/document.xml')
        
        tree = ET.fromstring(xml_content)
        
        namespaces = {'w': 'http://schemas.openxmlformats.org/wordprocessingml/2006/main'}
        
        texts = []
        for paragraph in tree.findall('.//w:p', namespaces):
            para_text = []
            for run in paragraph.findall('.//w:r', namespaces):
                text_elem = run.find('w:t', namespaces)
                if text_elem is not None and text_elem.text:
                    para_text.append(text_elem.text)
            if para_text:
                texts.append(''.join(para_text))
                
        return '\n'.join(texts)
    except Exception as e:
        return f"Error: {str(e)}"

if __name__ == "__main__":
    path = sys.argv[1]
    print(extract_text_from_docx(path))
